### DISCUSSION TITLE
WIFI streaming example not working in combination with loco positioning deck

### DISCUSSION THREAD

**Discussion by BNoordam** [bitcraze/discussions#1060] [Q&A] [answered]
Hello,

I am trying to implement the WIFI streaming example on the AI deck while the loco positioning deck is also attached. Via the virtual machine (getting started with develoment tutorial) I altered the aideck.c file such that there are no conflicts between the decks. Than I build and flashed the altered firmware according to the same tutorial. I checked with a python script that both decks were recognized. I also successfully flashed the GAP8 bootloader with the Olimex. However, when I try to flash the WIFI streaming example over the radio, the flashing gets stuck at 0%. I flashed the crazyflie again with just the AI deck connected via the cfclient. After that, the WIFI streaming example was possible to flash over the radio, but this caused the loco positioning deck to not be recognized anymore, I guess because the aideck.c file was overwritten in this progress. Also the WIFI streaming example only worked when I keep the loco positioning deck detached.  Is the problem because of the altered aideck.c file, or do I make a mistake with flashing?

**Comment by knmcguire** [comment]
Hi, So as I understand correctly, there is not perse an issue with the wifi streaming example but you fail to flash the streaming example of the aideck? Perhaps you should rewrite the title of this.

We have seen reports of flashing failing due to the addition of another deck. Does flashing work without the loco positioning deck? also, when you said 'Also the WIFI streaming example only worked when I keep the loco positioning deck detached', are you also referring to flashing or just the running of the wifi example itself?

**Reply by knmcguire** [reply]
I also just double-checked this internally, and the fact that the chip select of the loco positioning system and boot pin is connected will have an effect of flashing at the bootpin becomes important and shouldn't be meddled with (which the CS pin does). More specific than that I should ask one of my colleagues to help out, but the current instentive was: flashing won't work with both decks attached but once it is flashed it should be able to run

https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#deck-pin-allocation

**Reply by BNoordam** [reply]
Thanks for your reply.
Indeed the combination of two decks made it not possible to flash. I performed all of the flashing operations again with just the AI deck attached, and it was able to flash and perform the WIFI streaming example. However, when I attach the loco positioning deck the WIFI streaming example is no longer recognized by my WIFI. The LEDs of the AI deck also show a different behavior when the loco positioning deck is attached. Only LED2 is blinking rapidly, instead of LED1 blinking rapidly and LED2 slowly.

**Reply by knmcguire** [reply]
Yeaah... that means that the gap8 is set into bootloader mode... that's not good for sure. Seems there is more to be done to make them compatible. I'll forward this to my colleagues that know more these pins!

**Reply by BNoordam** [reply]
Alright, thanks for the help. It surprises me that in #854, where the workaround was proposed, the author did not have any problems with WIFI related tasks. Is it worth a shot to try some older firmware of the crazyflie?

**Reply by evoggy** [reply, marked as answer]
My best guess would be that the CS pin for the LPS deck is interfering with the BOOT pin. If the boot pin is pulled high at reset the ESP32 will go into bootloader mode, this would cause the described behavior above (the GAP8 not exiting bootloader and just blinking). One way to test this would be to delay the initialization of the LPS deck driver, giving the ESP32 time to boot up. After which point you can freely use the BOOT pin for CS to the LPS deck. This could be achieved with just sleeping a bit before initializing the driver.

**Reply by BNoordam** [reply]
Thanks alot, was indeed the problem. I put a delay of three seconds in locodeck.c at the beginning where the deck is initialized, and after it works perfectly.
